TSC/2021/04-30.md

```diff
@@ -0,0 +1,81 @@
+# Dissolve: Technical Steering Meeting
+### Fri 30/04/2021 13:30 - 14:30
+### Zoom
+
+### Attending
+
+- Tristan Youngs (TY)
+- Adam Washington (AW)
+- Stephen Smith (SS)
+- Anders Markvaardsen (AM)
+
+#### Apologies
+
+- Daniel Nixon (DN)
+- Lamar Moore (LM)
+
+### Agenda
+
+1. Sprint Review, weeks 16-17
+2. Parallelism Update
+3. Benchmarking
+4. GUI - Charts
+5. Applying Dissolve to a Large System
+6. AOB
+
+## 1. Sprint Review, weeks 16-17
+- TY noted that some significant PRs had been merged during the past sprint, most notably from SS and his work on multithreading.
+- Other major PRs from AW (Nix packaging, SpeciesTab modernisation) and TY (Bragg scattering, Array modernisation) are ongoing at this time.
+- AW noted that SnapCraft have redefined some things in their spec, and now only allow import of libs from specific directories. This causes issues for the ongoing work on the Nix packaging to the extent where some hacking of the system is going to be required.
+- TY stated that, since the idea of packaging via Nix was originally proposed, other native packages have been realised, and so the impact of the Nix build is lower than it once was.
+- For the system where it was envisaged Nix would be most useful (IDAaaS) TY said that a native build of Qt 5.15.2 and Dissolve was about to be integrated into IDAaaS by Franz Lang, using scripts TY had already written.
+- In light of this, AW, TY, and SS agreed that Nix packaging should be suspended indefinitely, but that Docker containers and a SnapCraft package would be relatively straightforward to achieve.
+
+ACTION: AW to add relevant issues on GitHub.
+
+- AM Asked for clarification on the difference between Docker and Snap, to which AW explained that Snap doesn't use a VM, whereas Docker does, and that from the user's perspective a Snap acts like a single binary container, and which may be potentially stored on a central repository allowing automatic package updates on user's machines.
```

Comment:
This is not quite correct, Docker and Snap are essentially both containerisation platforms. Docker using kernel primitives such as namespaces, Snap using the likes of AppArmour and chroots.

I'm also not sure what a Docker image gains us, Docker is entirely the wrong solution to packaging Dissolve.

Reply:
Possible that I mis-quoted / recalled Adam here, but I take your point re Docker.